src/bp/common/controls.ts

```diff
@@ -0,0 +1,120 @@
+import { FormDynamicOptions } from 'botpress/sdk'
+
+export interface FormOption {
+  value: any
+  label: string
+  related?: any
+}
+
+export interface MoreInfo {
+  /** Content of the popover when hovering the icon */
+  label: string
+  /** An URL with additional documentation */
+  url?: string
+}
+
+export enum ControlType {
+  String = 'string',
+  Boolean = 'boolean',
+  Number = 'number',
+  Array = 'array',
+  /** Display a dropdown menu with a list of options  */
+  Enum = 'enum',
+  /** Display an upload component so users can pick a file from their filesystem */
+  File = 'file',
+
+  /** Display a custom component from an override field or from a custom module */
+  Component = 'component'
+}
+
+export interface BaseControl {
+  /** Label displayed on top of the control */
+  title: string
+  /** Display an help icon next to the control */
+  moreInfo?: MoreInfo
+  defaultValue?: any
+  /** When true, the user can provide a different value for this property for each language supported by the bot */
+  translated?: boolean
+  /** Indicates that this field must be filled */
+  required?: boolean
+  /** Text to display in the control when no input is provided */
+  placeholder?: string
+  /** The name of the section where this property will be */
+  section?: string
+  // TODO: Unsure about these 3, keeping them temporarily
+  fields?: any
+  onClick?: any
+  group?: any
+}
+
+export interface ControlBoolean extends BaseControl {
+  type: ControlType.Boolean
+  defaultValue?: boolean
+}
+
+export interface ControlEnum extends BaseControl {
```

Comment:
Yup, a list of elements. Trying to represent controls by the type of data you want to handle, and not the control you want to display. For example, the enum uses a Select component, but it switches to MultiSelect when you set the property `multiple` to true. 

Basically, you don't have to inspect the code to choose the right component, you just fill in the properties using the auto code completion and we handle the components behind the scene

Reply:
Is there a what we can add some documentation. So not only you knows how that stuff works

Reply:
Well the point of those modifications are exactly so no documentation is needed and the code talks for itself. Typings were simplified and you have autocompletion when you use it. There's no point for additional documentation since it's not being used by outside users yet